core/domain/suggestion_services.py

```diff
@@ -80,12 +80,12 @@
 ] = {
     feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT: (
         lambda suggestion: suggestion.change.translation_html),
-    # Here, change is of type BaseChange and all attributes on BaseChange are
-    # created dynamically except cmd, and the type of all dynamically created
-    # attributes are considered as string (str) type. So, question_dict
-    # is also considered as string type but here we are using it as a Dict
-    # type which causes MyPy throws an error. Thus to avoid the error, we
-    # used ignore here.
+    # Here we use MyPy ignore because change is of type BaseChange and all
```

Comment:
This should be done using subclasses of BaseChange.

Reply:
This issue is already fixed in https://github.com/oppia/oppia/pull/15958

Reply:
No -- as far as I know, we cannot remove this ignore. reason:

In dict 
https://github.com/oppia/oppia/blob/68bb129c7fe773b989670a59162534b8893789bb/core/domain/suggestion_services.py#L78
we are defining different lambda functions with different suggestions. So, to allow every suggestion we have to use BaseSuggestion and in baseSuggestion change is defined as BaseChange and base change does not contain extra methods.

Reply:
I think i need to explain a bit more here:

> In this Dict, we are defining different lambda functions with different suggestions. So, to allow every suggestion we have to use BaseSuggestion and in baseSuggestion change is defined as BaseChange and base change does not contain extra methods/attributes.

So, these extra attributes have type str because they are created dynamically in BaseChange class. that's why question_dict is considered as a type of str. Becasue question_dict is created dynamically in BaseChange class.